### PR TITLE
Update talent plan 2.0

### PR DESCRIPTION
**Pull request by Soline324**
Talent Plan is meeting its 2.0 version. More courses related to open source, programming language, distributed database and deep dive into tidb ecosystems will be added. That's why I create this pull request. Changes include:

* update `README.md`
* add three new courses related to open source collaboration, which are:
  * TP 101: Introduction to open source software
  * TP 102: How to use Git and GitHub
  * TP 103: Build a welcoming open source community
* add `CONTRIBUTING.md`
* add `Code_of_conduct.md`
* reorg all the public courses

Other courses will be updated by my fellows as soon as possible, which are:

- [ ] add a brief description to `tinysql`,  @winoros 
- [ ] archive `tidb` folder, @winoros
- [ ] add a brief description to`tinykv`, @Connor1996
- [ ] move `rust` to `courses` dictionary 
- [ ] move `dss` to `courses` dictionary

@zz-jason PTAL

### CODE REVIEW

**Comment by Soline324** [comment]
ping @brson

File: courses/tp101-intro-to-oss.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+
+
+
+
+
+
+
+
+
```
Comment (Soline324) [inline courses/tp101-intro-to-oss.md:85]:
```suggestion
```

File: courses/tp102-how-to-use-git-github.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+
+
+
+
+
+
+
+
+
```
Comment (Soline324) [inline courses/tp102-how-to-use-git-github.md:104]:
```suggestion

```

File: courses/tp103-open-source-community.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+
+
+
+
+
+
+
+
+
```
Comment (Soline324) [inline courses/tp103-open-source-community.md:103]:
```suggestion

```

File: README.md [hunk 4]
```diff
@@ -1,50 +1,57 @@
...
 
-The courses primarily consist of projects (or "labs") where coding problems are
-presented, along with a partial implementation or API description, and a test
-suite.
+Talent Plan is an open source training program initiated by PingCAP. It aims to create or combine some open source learning materials for people interested in open source, distributed systems, Rust, Golang, and other infrastructure knowledges. As such, a series of courses focused on open source collaboration, rust programming, distributed database and systems are provided.
 
-Each course is developed independently, so they vary in their presentation and
-their expectations from course-takers. See the individual course documentation
-for details.
+Note:
```
Comment (zz-jason) [inline README.md:7]:
```suggestion
> Note:
```

File: README.md [hunk 5]
```diff
@@ -1,50 +1,57 @@
...
+
+Two courses are included in this series, which are:
+
+- TP 301: TinyKV, a distributed key value database in Go(WIP) 
+- TP 302: TinySQL, a distributed relational database in Go(WIP)
+
+### Series 4: Deep Dive into TiDB Ecosystems 
+
+- Deep Dive into TiDB(WIP)
+- Deep Dive into TiKV(WIP)
```
Comment (zz-jason) [inline README.md:39]:
```suggestion
- TP 401: Deep Dive into TiDB(WIP)
- TP 402: Deep Dive into TiKV(WIP)
```

File: README.md [hunk 6]
```diff
@@ -1,50 +1,57 @@
...
-[Percolator]: https://storage.googleapis.com/pub-tools-public-publication-data/pdf/36726.pdf
+- [TP 201: Practical Networked Applications in Rust](rust/README.md). A series of projects that incrementally develop a single Rust project from the ground up into a high-performance, networked, parallel and asynchronous key/value store. Along the way various real-world and practical Rust development subject matter are explored and discussed.
+
+- [TP 202: Distributed Systems in Rust](dss/README.md). Adapted from the [MIT 6.824](http://nil.csail.mit.edu/6.824/2017/index.html) distributed systems coursework, this course focuses on implementing important distributed algorithms, including the [Raft](https://raft.github.io/) consensus algorithm, and the [Percolator](https://storage.googleapis.com/pub-tools-public-publication-data/pdf/36726.pdf) distributed transaction protocol.
+
+### Series 3: Distributed Database 
+
+Two courses are included in this series, which are:
+
+- TP 301: TinyKV, a distributed key value database in Go(WIP)
```
Comment (zz-jason) [inline README.md:33]: should we link them to the repository of `tinysql` and `tinykv`?

Comment (Soline324) [reply README.md:33]: It seems to work, what do you think? @Connor1996 @winoros

File: README.md [hunk 7]
```diff
@@ -1,50 +1,57 @@
...
+- [TP 201: Practical Networked Applications in Rust](rust/README.md). A series of projects that incrementally develop a single Rust project from the ground up into a high-performance, networked, parallel and asynchronous key/value store. Along the way various real-world and practical Rust development subject matter are explored and discussed.
+
+- [TP 202: Distributed Systems in Rust](dss/README.md). Adapted from the [MIT 6.824](http://nil.csail.mit.edu/6.824/2017/index.html) distributed systems coursework, this course focuses on implementing important distributed algorithms, including the [Raft](https://raft.github.io/) consensus algorithm, and the [Percolator](https://storage.googleapis.com/pub-tools-public-publication-data/pdf/36726.pdf) distributed transaction protocol.
+
+### Series 3: Distributed Database 
+
+Two courses are included in this series, which are:
+
+- TP 301: TinyKV, a distributed key value database in Go(WIP) 
+- TP 302: TinySQL, a distributed relational database in Go(WIP)
```
Comment (Soline324) [inline README.md:34]:
```suggestion
- [TP 301: TinyKV, a distributed key value database in Go](https://github.com/pingcap-incubator/tinykv) 
- [TP 302: TinySQL, a distributed relational database in Go](https://github.com/pingcap-incubator/tinysql)
```

File: courses/README.md [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+
+Here we list all public courses of Talent Plan, including:
+
+- [TP 101: Introduction to open source software](tp101-intro-to-oss.md)
+- [TP 102: How to use Git and GitHub](tp102-how-to-use-git-github.md)
+- [TP 103: Build a welcoming community](tp103-open-source-community.md)
+- [TP 201: Practical Networked Applications in Rust](rust/README.md)
+- [TP 202: Distributed Systems in Rust](dss/README.md)
+- TP 301: TinyKV, a distributed key value database in Go(WIP) 
+- TP 302: TinySQL, a distributed relational database in Go(WIP)
```
Comment (Soline324) [inline courses/README.md:11]:
```suggestion
- [TP 301: TinyKV, a distributed key value database in Go](https://github.com/pingcap-incubator/tinykv) 
- [TP 302: TinySQL, a distributed relational database in Go](https://github.com/pingcap-incubator/tinysql)
```

File: README.md (hunk 7 above)
Comment (Soline324) [reply README.md:34]: Maybe we can link `tinysql` and `tinykv`to the repository in `pingcap-incubator` org directly, still we'd better add some brief description of these two courses in `REAMDE.md`. What do you think?

@zz-jason @Connor1996 @winoros

File: README.md (hunk 6 above)
Comment (Connor1996) [reply README.md:33]: so, we do not need to move it into talent-plan repo?

File: README.md (hunk 7 above)
Comment (Connor1996) [reply README.md:34]: oh, that would be better

File: README.md (hunk 6 above)
Comment (Soline324) [reply README.md:33]: yes, we can link it to the repository in `pingcap-incubator` org directly.

File: README.md (hunk 7 above)
Comment (Soline324) [reply README.md:34]: ok, then I'll commit this suggestion. Could you please add a brief description to tinysql and tinykv courses?

@Connor1996 @winoros

**Pull request merged**